### ISSUE TITLE
Wow huge work

### ISSUE THREAD

**Issue by iSuslov** [Mbodin/murder-generator#5]
Just wanna say that I'm impressed. Thank you for doing this. One question, why did you choose OCaml? Is it something special about this language for such a project?

**Comment by Mbodin** [comment]
Hi! Thanks for the feedback ☺ I should try to get some time to continue working on this project: I didn’t got back to it since the lockdown.

I just choose OCaml because I’m used to it, really ☺ The fact that there exists a compiler to JavaScript is a bonus, as one can try the tool out without installing anything. It also feels to me that the way I designed the solver better fits a functional style. But I guess that any other programming language could have done the job ☺

**Comment by blakeyoung81** [comment]
you should make a way to edit the scenario after it has been built. I'd love to use the generate scenario as a template

**Comment by Mbodin** [comment]
> you should make a way to edit the scenario after it has been built. I'd love to use the generate scenario as a template

I fully agree. I currently have very few time to actually do much on this project, but the discussion is interesting ☺
I could see several ways in which I could improve the project to make it able to use generated scenario as a template:
- I could generate it in a format that would made edition possible or easy. I plan to do a LaTeX extraction, for instance. Do you have another in mind? This could enable someone to customise the scenario, but possibly with a lot of effort.
- I could enable to drag and drop the generated events or changing their time (I have some ideas of interfaces, but I need to try them out), as well as enabling to change the text associated to an event. This could help making small changes to a scenario once generated (and in particular, could help maintain some coherence along the scenario), but it might be slightly limited in what it can do.
- I don’t know how this could be done, but there might be a way to create changes that would be understood by the solver. This would enable to for instance change something (like the time of an event, or some property of a character, or even some text) out of a generated scenario, and be able to tell the solver “now complete with something else”. This would probably need a lot of work :-\
- As I understand it, the three items above still follow the logic of “a scenario is a bunch of events happening to different characters, and each can have some effects on why characters think or behave as they currently do”. Would an edition that does not fit within this model still be interesting? (Knowing that there would be almost no hope to make it compatible with the solver.)

@blakeyoung81: Which of these ways of altering the scenario would be interesting to you?
